plugins/modules/aci_static_binding_to_epg.py

```diff
@@ -367,13 +370,17 @@ def main():
 
     if encap_id is not None:
         if encap_id not in range(1, 4097):
-            module.fail_json(msg='Valid VLAN assigments are from 1 to 4096')
+            module.fail_json(msg='Valid VLAN assignments are from 1 to 4096')
         encap_id = 'vlan-{0}'.format(encap_id)
 
     if primary_encap_id is not None:
-        if primary_encap_id not in range(1, 4097):
-            module.fail_json(msg='Valid VLAN assigments are from 1 to 4096')
-        primary_encap_id = 'vlan-{0}'.format(primary_encap_id)
+        try:
+            primary_encap_id = int(primary_encap_id)
+            if isinstance(primary_encap_id, int) and primary_encap_id in range(1, 4097):
```

Comment:
What happens if primary_encap_id == 4098? I think you are not printing an error like it was before.

Reply:
Can we add a test for that?